src/cargo/core/compiler/mod.rs

```diff
@@ -1325,6 +1327,36 @@ fn build_base_args(
         opt(cmd, "-C", "incremental=", Some(dir));
     }
 
+    let pkg_hint_mostly_unused = match hints.mostly_unused {
+        None => None,
+        Some(toml::Value::Boolean(b)) => Some(b),
+        Some(v) => {
+            bcx.gctx.shell().warn(format!(
+                "ignoring unsupported value type ({}) for 'hints.mostly-unused', which expects a boolean",
```

Comment:
Can this warning (and the ones below) include the package id? Otherwise I don't think the user would be able to know where this is coming from.